src/table.ts

```diff
@@ -1904,14 +1904,17 @@ class Table extends common.ServiceObject {
         ? optionsOrCallback
         : ({} as InsertRowsOptions);
     const callback =
-      typeof optionsOrCallback === 'function'
-        ? optionsOrCallback
-        : (cb as InsertRowsCallback);
+      typeof optionsOrCallback === 'function' ? optionsOrCallback : cb;
 
-    this._insertAndCreateTable(rows, options).then(
-      resp => callback(null, resp),
-      err => callback(err, null)
-    );
+    const promise = this._insertAndCreateTable(rows, options);
+    if (callback) {
+      promise.then(
+        resp => callback(null, resp),
+        err => callback(err, null)
+      );
+    } else {
+      return promise.then(r => [r]);
```

Comment:
It's sort of odd to me that the Promise returns an array, but the callback isn't called with one. This seems to maintain the method definition though.

Reply:
Or it could be typescript having trouble determining the return type.
I'm not sure if there's a test for the callback version's return type.